app/controllers/user_controller.php

```diff
@@ -91,29 +112,137 @@ public function profile()
-            $user->lastname = Param::get('lastname');
-            $user->username = Param::get('username');
-            $user->email = Param::get('email');
-                    
-            try {
-                $user->edit();
-            } catch (ValidationException $e) {
-                $page = 'profile';
-            }    
-            break;
-        default:
-            throw new NotFoundException("{$page} not found");
-            break;
+            case 'profile':
+                break;
+            case 'success_update':
+                $user->firstname = Param::get('firstname');
+                $user->lastname = Param::get('lastname');
+                        
+                try {
+                    $user->edit();
+                } catch (ValidationException $e) {
+                    $page = 'profile';
+                }    
+                break;
+            default:
+                throw new NotFoundException("{$page} not found");
+                break;
         }
 
         $this->set(get_defined_vars());
         $this->render($page);
     }
+
+    public function status()
+    {
+        if ($_SESSION['usertype'] == ADMIN) {
+            $user = User::getAll();        
+
+            if ($user) {
+                $current_page = max(Param::get('page'), SimplePagination::MIN_PAGE_NUM);
+                $pagination = new SimplePagination($current_page, self::MAX_ITEMS_PER_PAGE);
+                $other_threads = array_slice($user, $pagination->start_index + SimplePagination::MIN_PAGE_NUM);
+                $pagination->checkLastPage($other_threads);
+                $page_links = Pagination(count($user), self::MAX_ITEMS_PER_PAGE, $current_page, self::ADJACENT_TO_CURRENT);
+                $user = array_slice($user, $pagination->start_index - 1, $pagination->count);
+            }
+
+            $count = count($page_links);
+            $this->set(get_defined_vars());
+        } else {
+            redirect('/user/home');
+        }
+    }
+
+    public function edit_status()
+    {
+        $users = new User();
+        $user_id = Param::get('user_id');
+        $users->user_id = $user_id;
+        $user = User::get($user_id);
+        $users->current_status = $user['status'];
+        
+        $update = $users->editStatus();
+
+        $this->set(get_defined_vars());
+        redirect("/user/status?page={$_SESSION['current_page']}");
+    }
+
+    public function change_password()
+    {
+        $user = new User();
+        $page = Param::get('page_next', 'change_password');
+
+        switch ($page) {
+            case 'change_password':
+                break;
+            case 'edit_success':
+                $user->old_password = Param::get('old_password');
+                $user->password = Param::get('password');
+                $user->confirm_password = Param::get('confirm_password');
+                        
+                try {
+                    $change_password = $user->changePassword();
```

Comment:
please add restriction po if the current user is the one changing his own password :bow: